### PR TITLE
Feature/exchanges management

### PR DESCRIPTION
**Pull request by micborens**
## Proposed Changes
After installation of a RabbitMQ Cluster, i realized that i need to manage exchange (create or delete) automatically.
This feature was not present in the repository, so I developed it.
The goal is to be able to declare and delete one or more exchanges.

## Types of Changes
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation (correction or otherwise)

## Checklist
- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in related repositories

### CODE REVIEW

File: resources/exchanges.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#
+
+actions :declare, :delete
+default_action :declare
+
+attribute :exchange, :kind_of => String, :name_attribute => true
+attribute :type, :kind_of => String, :name_attribute => true
+attribute :durable, :kind_of => [TrueClass, FalseClass], :name_attribute => true
+attribute :auto_delete, :kind_of => [TrueClass, FalseClass], :name_attribute => true
+attribute :vhost, :kind_of => String, :name_attribute => true
```
Comment (devsibwarra) [inline resources/exchanges.rb]: Pretty sure you only want one of these attributes marked as a `name_attribute` for the LWRP

File: providers/exchanges.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include Opscode::RabbitMQ
+
+use_inline_resources
+
+def exchanges_exists?(vhost, exchange)
+    cmd = "rabbitmqctl -q list_exchanges -p #{vhost}| grep #{exchange}"
```
[the remark this answers is not in the document]
Comment (devsibwarra) [reply providers/exchanges.rb]: From the `rabbitmqctl --help` output
```
list_exchanges [-p <vhost>] [<exchangeinfoitem> ...]
<exchangeinfoitem> must be a member of the list [name, type, durable, auto_delete, internal, arguments, policy].
```

I'd update this command to use `rabbitmqctl -q list_exchanges -p #{vhost} name` in order to make the grep anchoring better.

```
# rabbitmqctl -q list_exchanges -p / | grep exchange_one
test_exchange_one	direct
# rabbitmqctl -q list_exchanges -p / name | grep exchange_one
test_exchange_one
# rabbitmqctl -q list_exchanges -p / name | grep '^test_exchange_one$'
test_exchange_one
# rabbitmqctl -q list_exchanges -p / name | grep '^exchange_one$'
#
```

File: resources/exchanges.rb (hunk 1 above)
Comment (michaelklishin) [reply resources/exchanges.rb]: Exchange name makes most sense as the name attribute. Of course there cannot be 4 named attributes.

File: providers/exchanges.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (micborens) [reply providers/exchanges.rb]: i will try and update cmd to have better match with exchanges. thx a lot for your feedback

**Comment by michaelklishin** [comment]
Closing due to inactivity. We'd consider a similar PR that addresses the points discussed above. Thank you for your time, @micborens!

**Pull request closed without merging**